### PR TITLE
Fix segfault on shortening readonly label by allocating memory

### PR DESCRIPTION
**Pull request by livibetter**
While I was trying `demo` with smaller terminal columns <64, the second example caused a segmentation fault.

It's due to the shortening of the label:

```c
  if (maxstrlen - 10 > 10 && strlen(label) > maxstrlen - 10) {
    label[maxstrlen - 10] = 0;
  }
```

The `label` from `test/progressbar_demo.c` is a literal string and read-only. If the demo passes:

```c
char label[] = "long label...";
```

This won't cause any trouble, but I don't think it would be convenient to use this library, therefore I submit this patch.

### CODE REVIEW

**Comment by livibetter** [comment]
@doches I did a force update, let me know if anything doesn't fit the style or if I create more bugs.

File: lib/progressbar.c [hunk 1]
```diff
@@ -54,7 +55,12 @@ progressbar *progressbar_new(char *label, unsigned long max)
 
 void progressbar_update_label(progressbar *bar, char *label)
 {
-  bar->label = label;
+  if (bar->label) {
+    free(bar->label);
```
Comment (marcomorain) [inline lib/progressbar.c]: Freeing a null pointer is safe - no need for the `if` here.

File: lib/progressbar.c [hunk 2]
```diff
@@ -54,7 +55,12 @@ progressbar *progressbar_new(char *label, unsigned long max)
...
+  }
+  bar->label = calloc(strlen(label)+1, sizeof(char));
+  label = strncpy(bar->label, label, strlen(label));
```
Comment (marcomorain) [inline lib/progressbar.c]: You should use `strdup(3)` rather than manually allocating and copying memory.

Comment (livibetter) [inline lib/progressbar.c]: @marcomorain thanks for the suggestion, I will make the changes.

@doches don't merge yet.

**Comment by livibetter** [comment]
I didn't want to edit the commit and it's actually better to stack up another commit, which also changes others to use `strdup` (per @marcomorain's suggestion) not just the code for this issue.

**Comment by doches** [comment]
So, @livibetter -- can I close out this PR without a merge and call it a day? It's been a useful conversation either way, but I think @marcomorain's commits supersede what you were fixing here.

**Comment by livibetter** [comment]
@doches I wasn't aware of those, closing.

**Pull request closed without merging**